examples/ex7_code_insertion_with_extension.py

```diff
@@ -86,7 +82,23 @@ async def add_and_return_segment(elf_resource: Resource, vaddr: int, size: int)
     file_segments = await elf_resource.get_descendants_as_view(
         ElfProgramHeader, r_filter=ResourceFilter(tags=(ElfProgramHeader,))
     )
-    return [seg for seg in file_segments if seg.p_vaddr == vaddr].pop()
+    segment = [seg for seg in file_segments if seg.p_vaddr == vaddr].pop()
+
+    # Carve out a child of the new segment where we can store the code for our new function.
+    code_region = CodeRegion(segment.p_vaddr + GHIDRA_PIE_OFFSET, segment.p_filesz)
```

Comment:
Nitpick, but i would break the portion adding FreeSpace to a new function for readability.